tests/report_generator_test.py

```diff
@@ -0,0 +1,74 @@
+"""Report Generator Test"""
+
+import unittest
+
+from pipeline_dp.report_generator import ReportGenerator
+from pipeline_dp.dp_engine import AggregateParams, Metrics
+
+class ReportGeneratorTest(unittest.TestCase):
+  def test_report_empty(self):
+    ReportGenerator(None).report()  # no errors
+
+  def test_report_params(self):
+    params = AggregateParams(
+      max_partitions_contributed=2,
+      max_contributions_per_partition=1,
+      low=1,
+      high=5,
+      metrics=[Metrics.PRIVACY_ID_COUNT,Metrics.COUNT,Metrics.MEAN],
+    )
+    ReportGenerator(params).report() # no errors
+
+  def test_report_aggregate(self):
+    params1 = AggregateParams(
+      max_partitions_contributed=3,
+      max_contributions_per_partition=2,
+      low=1,
+      high=5,
+      metrics=[Metrics.PRIVACY_ID_COUNT,Metrics.COUNT, Metrics.MEAN],
+    )
+    params2 = AggregateParams(
+      max_partitions_contributed=1,
+      max_contributions_per_partition=3,
+      low=2,
+      high=10,
+      metrics=[Metrics.VAR,Metrics.SUM,Metrics.MEAN],
+      public_partitions = list(range(1,40)),
+    )
+    engine_stub = DPEngineStub()
+    engine_stub.aggregate(params1)
+    engine_stub.aggregate(params2)
+    for _, report_aggregate in enumerate(engine_stub.report_generators):
+      print(report_aggregate.report()) # no errors
+
+class DPEngineStub():
+  """Stub DPEngine for testing."""
+  def __init__(self):
+    self.report_generators = []
+
+  def _add_report_stage(self, text):
+    self.report_generators[-1].add_stage(text)
+
+  def aggregate(self, params):
+    self.report_generators.append(ReportGenerator(params))
+    self._add_report_stage(f"Clip values to {params.low, params.high}")
+    self._add_report_stage(
```

Comment:
For simplicity I'd recommend to put just a few add_stage: with str and a function and check that text in them are in output 
(something like:
s = "..."
report_generator.add_stage(s)
...
check that s in report.

And it's better to have some short/simple text, no need to reproduce how it will be used. Could you please put it in a separate test